File: client/src/app/modules/tiers/services/special-offer.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/appRest.service';
import { SpecialOffer } from '../../tiers/model/SpecialOffer';

@Injectable({
  providedIn: 'root'
})
export class SpecialOfferService extends AppRestService<SpecialOffer>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getSpecialOffers() {
    return this.getList(new HttpParams(), "special-offers");
  }

}

File: src/main/java/com/jss/jssbackend/modules/tiers/service/SpecialOfferService.java
package com.jss.jssbackend.modules.tiers.service;

import java.util.List;

import com.jss.jssbackend.modules.tiers.model.SpecialOffer;

public interface SpecialOfferService {
    public List<SpecialOffer> getSpecialOffers();

    public SpecialOffer getSpecialOffer(Integer id);
}
